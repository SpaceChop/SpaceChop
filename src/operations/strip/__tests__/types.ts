import StripConfig, { StripConfig as Config } from '../types';

describe('should validate strip config', () => {
  describe('should throw on invalid config', () => {
    it('require valid icc_profile', () => {
      const config = {
        icc_profile: 'icc_profile',
      };
      expect(() => StripConfig.check(config)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('should now throw on valid config', () => {
    it(`valid icc_profile`, () => {
      const config: Config = {
        icc_profile: true,
      };
      expect(() => StripConfig.check(config)).not.toThrow();
    });
  });
});
