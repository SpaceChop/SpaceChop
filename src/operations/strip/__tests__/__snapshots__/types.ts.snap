// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate strip config should throw on invalid config require valid icc_profile 1`] = `"Expected boolean, but was string"`;
